### ISSUE TITLE
Emoji classified as letter

### ISSUE THREAD

**Issue by djstrong** [mrabarnett/mrab-regex#473]
In Python 3.10.5, with regex 2022.4.24:
```
>>> import regex
>>> regex.match(r'^\p{LC}+$','😺')
<regex.Match object; span=(0, 1), match='😺'>
```
According to this: https://unicode.link/codepoint/1F63A-smiling-cat-face-with-open-mouth it has `So` category.

**Comment by mrabarnett** [comment]
Interesting.

```
>>> regex.match(r'^\p{Lc}+$','\N{SMILING CAT FACE WITH OPEN MOUTH}')
<regex.Match object; span=(0, 1), match='😺'>
>>> regex.match(r'^\p{So}+$','\N{SMILING CAT FACE WITH OPEN MOUTH}')
<regex.Match object; span=(0, 1), match='😺'>

```

So it's mishandling the `Lc` category somehow.

**Comment by djstrong** [comment]
```
>>> regex.match(r'^\p{LC}+$','😺')
<regex.Match object; span=(0, 1), match='😺'>
>>> regex.match(r'^\p{L}+$','😺')
>>> regex.match(r'^\p{Lu}+$','😺')
>>> regex.match(r'^\p{Ll}+$','😺')
>>> regex.match(r'^\p{Lt}+$','😺')
```
According to this: http://www.unicode.org/reports/tr44/#GC_Values_Table LC=Lu | Ll | Lt

**Comment by mrabarnett** [comment]
Fixed in regex 2022.7.9.

By the way, when building with GitHub Actions, "Run test on macos-latest" failed because, apparently, "The 'regex==2022.7.9' distribution was not found and is required by the application".

"Run test on windows-latest" and "Run test on ubuntu-latest" both passed, so it looks like a Mac OS problem.
